Using Nakayama functors to derive injective methods and install IndecomposableProjective/InjectiveObjects as CAP operations

kamalsaleh opened the pull request:
Particularly, with this commit I can ensure that any `InjectiveEnvelopeObject` of an object in `PSh` is a direct sum of objects in `IndecomposableInjectiveObjects( PSh ).`

depends on https://github.com/homalg-project/CAP_project/pull/1112

kamalsaleh committed:
change names of operations

1- RadicalInclusion -> RadicalInclusionOfPreSheaf
2- CoverElementByProjectiveObject -> CoverElementByIndecomposableProjectivePreSheaf
3- MorphismsFromDirectSumDecompositionOfProjectiveCover -> ProjectiveCoverObjectDataOfPreSheaf

kamalsaleh committed:
add Indecomposable(Projective/Injective)Objects as CAP methods using Nakayama functor

mohamed-barakat commented on gap/HomologicalMethods.gi:
```diff
@@ -126,50 +126,25 @@ InstallOtherMethodForCompilerForCAP( MorphismsFromDirectSumDecompositionOfProjec
         [ IsObjectInPreSheafCategory ],
 
-  MonomorphismIntoInjectiveEnvelopeObject );
+  EpimorphismFromSomeProjectiveObject );
```
There is a missing newline here

kamalsaleh committed:
remove operations

1- DualOf(Object/Morphism)InPreSheafCategory
2- MorphismsIntoDirectSumDecompositionOfInjectiveEnvelope
3- DirectSumDecompositionOfInjectiveObject
4- InjectiveEnvelope

kamalsaleh committed:
coding style

mohamed-barakat committed:
bumped version
